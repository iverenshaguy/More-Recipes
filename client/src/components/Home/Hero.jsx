import React from 'react';
import { InputGroup, InputGroupButton, Button, Form, Label, Input } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';

const Hero = props => (
  <div className="container-fluid" id="hero-image">
    <div className="row mx-auto" id="hero-image-text">
      <div className="col align-self-center">
        <div className="row justify-content-center align-items-center mb-3">
          <div className="col-12">
            <h3>Search for a Recipe</h3>
          </div>
          <div className="col-8 py-4">
            <Form id="search-form">
              <InputGroup size="lg">
                <Label className="sr-only" for="search">
                  Search for Recipes
                </Label>
                <Input id="search" placeholder="Curried Chicken Gravy..." />
                <InputGroupButton>
                  <Button type="submit" className="search-btn">
                    <FontAwesome name="search" />
                  </Button>
                </InputGroupButton>
              </InputGroup>
            </Form>
          </div>
        </div>
        <Button
          onClick={() => props.toggle('addRecipeModal')}
          className="btn-default btn-lg"
          id="home-add-recipe-btn"
          title="New Recipe"
        >
          Add a New Recipe
        </Button>
      </div>
    </div>
  </div>
);

Hero.propTypes = {
  toggle: PropTypes.func.isRequired,
};

export default Hero;
