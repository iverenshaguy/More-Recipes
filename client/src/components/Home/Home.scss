@import '../../assets/scss/_modules.scss';

.home {
  .navbar {
    position: relative;
  }

  #hero-image {
    height: 45vh;
    max-width: 100%;
    position: relative;
  }

  #hero-image-text {
    padding-top: 5vh;
    width: 100%;
    color: white;
    text-align: center;
  }

  #hero-search-form {
    margin-top: 2rem;
  }

  #hero-image button {
    background-color: $light-green;
    border: 1px solid $light-green;
    color: white;
    padding: 0.75rem 1.5rem;

    &:hover,
    &:focus,
    &:active {
      background-color: $green;
      border: 1px solid $green;
      color: white;
    }
  }
}

#home-add-recipe-btn {
  box-shadow: 0 !important;
  outline: none !important;

  &:hover,
  &:focus,
  &:active {
    cursor: pointer !important;
  }

  border-radius: 0;
}
