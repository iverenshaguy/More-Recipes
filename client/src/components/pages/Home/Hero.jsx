import React from 'react';
import PropTypes from 'prop-types';
import { AddRecipeBtn } from '../../shared/Buttons';
import SearchForm from '../../shared/Forms/SearchForm';

/**
 * @exports
 * @function Hero
 * @param {object} props
 * @returns {JSX} Hero
 */
const Hero = (props) => {
  const {
    handleSearch, handleAddRecipe, handleSearchInput, searchValue
  } = props;

  return (
    <div className="container-fluid" id="hero-image">
      <div className="row mx-auto" id="hero-image-text">
        <div className="col align-self-center">
          <div className="row justify-content-center align-items-center mb-3">
            <div className="col-12">
              <h3>Search for a Recipe</h3>
            </div>
            <div className="col-8 py-4">
              <SearchForm
                size="lg"
                handleSearch={handleSearch}
                handleSearchInput={handleSearchInput}
                searchValue={searchValue}
              />
            </div>
          </div>
          <AddRecipeBtn handleClick={handleAddRecipe} />
        </div>
      </div>
    </div>);
};

Hero.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  handleAddRecipe: PropTypes.func.isRequired,
  handleSearchInput: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired
};

export default Hero;
