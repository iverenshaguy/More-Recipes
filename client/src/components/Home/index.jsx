import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Hero from './Hero';
import RecipeItems from '../shared/RecipeItems';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  /**
   * @memberof Home
   * @returns {function} updateLocationState
   */
  componentWillMount() {
    this.props.updateLocationState('home');
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero toggle={this.props.toggleModal} />
        <RecipeItems />
      </div>
    );
  }
}

Home.propTypes = {
  toggleModal: PropTypes.func.isRequired,
  updateLocationState: PropTypes.func.isRequired,
};

export default Home;
