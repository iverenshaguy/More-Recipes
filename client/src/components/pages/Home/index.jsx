import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import setCurrentLocation from '../../../actions/location';
import { fetchTopRecipes, searchRecipes } from '../../../actions/recipes';
import Hero from './Hero';
import RecipeItems from '../../shared/RecipeItems';
import PreLoader from '../../shared/PreLoader';
import { multiRecipePropTypes } from '../../../helpers/proptypes';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired,
    ...multiRecipePropTypes
  };

  /**
   * @constructor
   * @memberof Home
   * @returns {nothing} - Returns nothing
   */
  constructor() {
    super();

    this.state = {
      title: 'TOP RECIPES',
      currentPage: 1,
      limit: 5,
      searchValue: ''
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.handleSearchInput = this.handleSearchInput.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  /**
   * @memberof Home
   * @return {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('home'));
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    let fetch;
    const { limit, searchValue } = this.state;

    if (this.state.title === 'TOP RECIPES') {
      fetch = fetchTopRecipes(page, limit);
    } else {
      fetch = searchRecipes(searchValue, page, limit);
    }

    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetch));
  }

  /**
   * @memberof Home
   * @param {event} event
   * @returns {state} home
   */
  handleSearchInput(event) {
    this.setState({
      searchValue: event.target.value
    });
  }

  /**
   * @memberof Home
   * @param {object} event
   * @returns {state} home
   */
  handleSearch(event) {
    event.preventDefault();

    const { searchValue, currentPage, limit } = this.state;
    this.setState({
      title: 'SEARCH RESULTS'
    }, () => this.props.dispatch(searchRecipes(searchValue, currentPage, limit)));
  }

  /**
   * @memberof Home
   * @returns {fragment} Recipe items
   */
  renderBody() {
    const { isFetching, recipes, metaData } = this.props;
    const { title } = this.state;

    if (isFetching) {
      return <PreLoader />;
    }

    return (
      <RecipeItems
        title={title}
        recipes={recipes}
        handlePageChange={this.handlePageChange}
        metaData={metaData}
      />
    );
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero
          dispatch={this.props.dispatch}
          handleSearchInput={this.handleSearchInput}
          handleSearch={this.handleSearch}
          searchValue={this.state.searchValue}
        />
        {this.renderBody()}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  recipes: state.recipes.recipes,
  metaData: state.recipes.metaData
});

export { Home as HomeComponent };

export default connect(mapStateToProps)(Home);
