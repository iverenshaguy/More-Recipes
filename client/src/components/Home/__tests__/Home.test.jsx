import React from 'react';
import Home from '../index';

describe('Home', () => {
  const toggleModalMock = jest.fn();
  const updateLocationStateMock = jest.fn();
  const wrapper = (
    <Home toggleModal={toggleModalMock} updateLocationState={updateLocationStateMock} />
  );

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const shallowWrapper = shallow(wrapper);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when fully mounted', () => {
    const mountedWrapper = mount(wrapper);
    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });

  it('calls component will mount when mounted', () => {
    const mountedWrapper = mount(wrapper);
    expect(updateLocationStateMock).toHaveBeenCalledWith('home');
    mountedWrapper.unmount();
  });
});
