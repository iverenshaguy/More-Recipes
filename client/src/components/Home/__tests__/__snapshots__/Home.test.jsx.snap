// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home"
>
  <Hero
    toggle={[Function]}
  />
  <RecipeItems />
</div>
`;

exports[`Home renders correctly when fully mounted 1`] = `
<Home
  toggleModal={[Function]}
  updateLocationState={[Function]}
>
  <div
    className="home"
  >
    <Hero
      toggle={[Function]}
    >
      <div
        className="container-fluid"
        id="hero-image"
      >
        <div
          className="row mx-auto"
          id="hero-image-text"
        >
          <div
            className="col align-self-center"
          >
            <div
              className="row justify-content-center align-items-center mb-3"
            >
              <div
                className="col-12"
              >
                <h3>
                  Search for a Recipe
                </h3>
              </div>
              <div
                className="col-8 py-4"
              >
                <Form
                  id="search-form"
                  tag="form"
                >
                  <form
                    className=""
                    id="search-form"
                  >
                    <InputGroup
                      size="lg"
                      tag="div"
                    >
                      <div
                        className="input-group input-group-lg"
                      >
                        <Label
                          className="sr-only"
                          htmlFor="search"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <label
                            className="sr-only  form-control-label"
                            htmlFor="search"
                          >
                            Search for Recipes
                          </label>
                        </Label>
                        <Input
                          id="search"
                          placeholder="Curried Chicken Gravy..."
                          tag="p"
                          type="text"
                        >
                          <input
                            className="form-control"
                            id="search"
                            placeholder="Curried Chicken Gravy..."
                            type="text"
                          />
                        </Input>
                        <InputGroupButton
                          tag="div"
                        >
                          <div
                            className="input-group-btn"
                          >
                            <Button
                              className="search-btn"
                              color="secondary"
                              tag="button"
                              type="submit"
                            >
                              <button
                                className="search-btn btn btn-secondary"
                                onClick={[Function]}
                                type="submit"
                              >
                                <FontAwesome
                                  name="search"
                                >
                                  <span
                                    aria-hidden={true}
                                    className="fa fa-search"
                                  />
                                </FontAwesome>
                              </button>
                            </Button>
                          </div>
                        </InputGroupButton>
                      </div>
                    </InputGroup>
                  </form>
                </Form>
              </div>
            </div>
            <Button
              className="btn-default btn-lg"
              color="secondary"
              id="home-add-recipe-btn"
              onClick={[Function]}
              tag="button"
              title="New Recipe"
            >
              <button
                className="btn-default btn-lg btn btn-secondary"
                id="home-add-recipe-btn"
                onClick={[Function]}
                title="New Recipe"
                type="button"
              >
                Add a New Recipe
              </button>
            </Button>
          </div>
        </div>
      </div>
    </Hero>
    <RecipeItems>
      <div
        className="container-fluid pb-5 background-grey page-margin"
        id="search-result-wrapper"
      >
        <div
          className="heading pt-4 text-center"
          id="search-result-heading"
        >
          <div
            className="col-xs-12 col-md-12 pt-3"
            id="title"
          >
            <h5>
              TOP RECIPES
            </h5>
          </div>
        </div>
        <hr />
        <div
          className="row pt-2 px-5"
          id="search-result"
        >
          <RecipeCard>
            <div
              className="col-lg-3 col-md-6 col-sm-12 col-xs-12 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <a
                    href="view-recipe.html"
                  >
                    <CardImg
                      alt="recipe"
                      className="img-fluid"
                      src="images/jollof-rice-img.jpg"
                      tag="img"
                      top={true}
                      width="100%"
                    >
                      <img
                        alt="recipe"
                        className="img-fluid card-img-top"
                        src="images/jollof-rice-img.jpg"
                        width="100%"
                      />
                    </CardImg>
                    <CardBody
                      className="card-block text-center pb-0"
                      tag="div"
                    >
                      <div
                        className="card-block text-center pb-0 card-body"
                      >
                        <h5
                          className="mb-1"
                        >
                          Jollof Rice and Chicken
                        </h5>
                        <CardText
                          className="mb-2"
                          tag="p"
                        >
                          <p
                            className="mb-2 card-text"
                          >
                            <span
                              className="text-muted"
                            >
                              Posted by 
                            </span>
                            @janesmith
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="card-block text-center recipe-details"
                      tag="div"
                    >
                      <div
                        className="card-block text-center recipe-details card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="clock-o"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-clock-o"
                              />
                            </FontAwesome>
                             45 minutes
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-0 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-0 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted d-inline card-text"
                          >
                            <i
                              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
                            />
                            A Bit Difficult
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="text-center text-muted px-2 recipe-rating"
                      tag="div"
                    >
                      <div
                        className="text-center text-muted px-2 recipe-rating card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="star"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-star"
                              />
                            </FontAwesome>
                             4.5
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="heart"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-heart"
                              />
                            </FontAwesome>
                             200
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              className="text-success"
                              name="eye"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-eye text-success"
                              />
                            </FontAwesome>
                             350
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                  </a>
                </div>
              </Card>
            </div>
          </RecipeCard>
          <RecipeCard>
            <div
              className="col-lg-3 col-md-6 col-sm-12 col-xs-12 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <a
                    href="view-recipe.html"
                  >
                    <CardImg
                      alt="recipe"
                      className="img-fluid"
                      src="images/jollof-rice-img.jpg"
                      tag="img"
                      top={true}
                      width="100%"
                    >
                      <img
                        alt="recipe"
                        className="img-fluid card-img-top"
                        src="images/jollof-rice-img.jpg"
                        width="100%"
                      />
                    </CardImg>
                    <CardBody
                      className="card-block text-center pb-0"
                      tag="div"
                    >
                      <div
                        className="card-block text-center pb-0 card-body"
                      >
                        <h5
                          className="mb-1"
                        >
                          Jollof Rice and Chicken
                        </h5>
                        <CardText
                          className="mb-2"
                          tag="p"
                        >
                          <p
                            className="mb-2 card-text"
                          >
                            <span
                              className="text-muted"
                            >
                              Posted by 
                            </span>
                            @janesmith
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="card-block text-center recipe-details"
                      tag="div"
                    >
                      <div
                        className="card-block text-center recipe-details card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="clock-o"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-clock-o"
                              />
                            </FontAwesome>
                             45 minutes
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-0 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-0 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted d-inline card-text"
                          >
                            <i
                              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
                            />
                            A Bit Difficult
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="text-center text-muted px-2 recipe-rating"
                      tag="div"
                    >
                      <div
                        className="text-center text-muted px-2 recipe-rating card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="star"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-star"
                              />
                            </FontAwesome>
                             4.5
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="heart"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-heart"
                              />
                            </FontAwesome>
                             200
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              className="text-success"
                              name="eye"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-eye text-success"
                              />
                            </FontAwesome>
                             350
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                  </a>
                </div>
              </Card>
            </div>
          </RecipeCard>
          <RecipeCard>
            <div
              className="col-lg-3 col-md-6 col-sm-12 col-xs-12 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <a
                    href="view-recipe.html"
                  >
                    <CardImg
                      alt="recipe"
                      className="img-fluid"
                      src="images/jollof-rice-img.jpg"
                      tag="img"
                      top={true}
                      width="100%"
                    >
                      <img
                        alt="recipe"
                        className="img-fluid card-img-top"
                        src="images/jollof-rice-img.jpg"
                        width="100%"
                      />
                    </CardImg>
                    <CardBody
                      className="card-block text-center pb-0"
                      tag="div"
                    >
                      <div
                        className="card-block text-center pb-0 card-body"
                      >
                        <h5
                          className="mb-1"
                        >
                          Jollof Rice and Chicken
                        </h5>
                        <CardText
                          className="mb-2"
                          tag="p"
                        >
                          <p
                            className="mb-2 card-text"
                          >
                            <span
                              className="text-muted"
                            >
                              Posted by 
                            </span>
                            @janesmith
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="card-block text-center recipe-details"
                      tag="div"
                    >
                      <div
                        className="card-block text-center recipe-details card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="clock-o"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-clock-o"
                              />
                            </FontAwesome>
                             45 minutes
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-0 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-0 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted d-inline card-text"
                          >
                            <i
                              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
                            />
                            A Bit Difficult
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="text-center text-muted px-2 recipe-rating"
                      tag="div"
                    >
                      <div
                        className="text-center text-muted px-2 recipe-rating card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="star"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-star"
                              />
                            </FontAwesome>
                             4.5
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="heart"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-heart"
                              />
                            </FontAwesome>
                             200
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              className="text-success"
                              name="eye"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-eye text-success"
                              />
                            </FontAwesome>
                             350
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                  </a>
                </div>
              </Card>
            </div>
          </RecipeCard>
          <RecipeCard>
            <div
              className="col-lg-3 col-md-6 col-sm-12 col-xs-12 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <a
                    href="view-recipe.html"
                  >
                    <CardImg
                      alt="recipe"
                      className="img-fluid"
                      src="images/jollof-rice-img.jpg"
                      tag="img"
                      top={true}
                      width="100%"
                    >
                      <img
                        alt="recipe"
                        className="img-fluid card-img-top"
                        src="images/jollof-rice-img.jpg"
                        width="100%"
                      />
                    </CardImg>
                    <CardBody
                      className="card-block text-center pb-0"
                      tag="div"
                    >
                      <div
                        className="card-block text-center pb-0 card-body"
                      >
                        <h5
                          className="mb-1"
                        >
                          Jollof Rice and Chicken
                        </h5>
                        <CardText
                          className="mb-2"
                          tag="p"
                        >
                          <p
                            className="mb-2 card-text"
                          >
                            <span
                              className="text-muted"
                            >
                              Posted by 
                            </span>
                            @janesmith
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="card-block text-center recipe-details"
                      tag="div"
                    >
                      <div
                        className="card-block text-center recipe-details card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="clock-o"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-clock-o"
                              />
                            </FontAwesome>
                             45 minutes
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-0 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-0 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted d-inline card-text"
                          >
                            <i
                              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
                            />
                            A Bit Difficult
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="text-center text-muted px-2 recipe-rating"
                      tag="div"
                    >
                      <div
                        className="text-center text-muted px-2 recipe-rating card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="star"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-star"
                              />
                            </FontAwesome>
                             4.5
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="heart"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-heart"
                              />
                            </FontAwesome>
                             200
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              className="text-success"
                              name="eye"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-eye text-success"
                              />
                            </FontAwesome>
                             350
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                  </a>
                </div>
              </Card>
            </div>
          </RecipeCard>
        </div>
      </div>
    </RecipeItems>
  </div>
</Home>
`;
